exo/networking/manual/manual_discovery.py

```diff
@@ -69,3 +68,45 @@ async def task_find_peers_from_config(self):
       await asyncio.sleep(1.0)
 
       if DEBUG_DISCOVERY >= 2: print(f"Current known peers: {[peer.id() for peer in self.known_peers.values()]}")
+
+  async def task_clean_up_peers_from_config(self):
+    if DEBUG_DISCOVERY >= 2: print("Starting task to clean up peers from config...")
+    while True:
+      peers_from_config = self._get_peers()
+      if peers_from_config:
+        peers_to_remove = [peer for peer in self.known_peers.keys() if peer not in peers_from_config]
+
+        for peer in peers_to_remove:
+          if DEBUG_DISCOVERY >= 2: print(f"{peer} is no longer found in the config but is currently a known peer. Removing from known peers...")
+          try: del self.known_peers[peer]
+          except KeyError: pass
+
+      await asyncio.sleep(1.0)
+
+  def _get_peers(self):
+    try:
+      current_mtime = os.path.getmtime(self.network_config_path)
```

Comment:
Isn't this blocking?
Blocking every 1 second is going to slow exo down significantly since it runs on one thread.

Perhaps we can move these operations to a thread pool executor?

Reply:
Yes, good catch. I updated all calls to `self.network_config_path` to run within a single thread pool executor. I think things should now be non-blocking and thread safe